contracts/Vault.vy

```diff
@@ -1166,20 +1168,35 @@ def report(_return: uint256) -> uint256:
         For approved strategies, this is the most efficient behavior.
         The Strategy reports back what it has free, then Vault "decides"
         whether to take some back or give it more. Note that the most it can
-        take is `_return`, and the most it can give is all of the remaining
+        take is `_gain`, and the most it can give is all of the remaining
         reserves. Anything outside of those bounds is abnormal behavior.
 
         All approved strategies must have increased diligence around
         calling this function, as abnormal behavior could become catastrophic.
-    @param _return
-        Amount Strategy has made on it's investment since its last report,
-        and is free to be given back to Vault as earnings
+    @param _gain
+        Amount Strategy has realized as a gain on it's investment since its
+        last report, and is free to be given back to Vault as earnings
+    @param _loss
+        Amount Strategy has realized as a loss on it's investment since its
+        last report, and should be accounted for on the Vault's balance sheet
     @return Amount of debt outstanding (iff totalDebt > debtLimit).
     """
 
     # Only approved strategies can call this function
     assert self.strategies[msg.sender].activation > 0
 
+    # We have a loss to report, do it before the rest of the calculations
+    if _loss > 0:
+        loss: uint256 = min(_loss, self.strategies[msg.sender].totalDebt)
```

Comment:
gain and loss are supposed to be mutually exclusive? meaning if i have a loss to report there shouldn't be any gain on that call, unless im missing something.  

Just thinking if we need to require if loss > 0 gain should equal 0 and vice versa, only issue is that the strategy could mess up and get stuck failing to not report

Reply:
I left it open here. I could see a scenario where you had both a gain and a loss to report e.g. you cashed in some reward tokens, but you lost some amount of underlying in the process. Of course, you could just mix the two together, but I think leaving it open makes it more flexible.